tidy3d/plugins/smatrix/component_modelers/terminal.py

```diff
@@ -0,0 +1,425 @@
+            h_cap_coords = [h_cap_field.x, h_cap_field.y, h_cap_field.z]
+
+            # Use the coordinates of h_cap since it lies on the same grid that the lumped resistor is snapped to
+            orth_index = np.argmin(
+                np.abs(h_cap_coords[port.injection_axis].values - port.center[port.injection_axis])
+            )
+            # Some sanity checks, tangent H field coordinates should be directly above
+            # and below the coordinates of the resistive sheet
+            assert orth_index > 0
+            assert (
+                h_cap_coords[port.injection_axis].values[orth_index]
+                < h_coords[port.injection_axis].values[orth_index]
+            )
+            assert (
+                h_coords[port.injection_axis].values[orth_index - 1]
+                < h_cap_coords[port.injection_axis].values[orth_index]
+            )
+
+            # Extract field just below and just above sheet
+            h1_field = h_field.isel({orth_component: orth_index - 1})
+            h2_field = h_field.isel({orth_component: orth_index})
+            h_field = h1_field - h2_field
+            # Extract cap field which is coincident with sheet
+            h_cap = h_cap_field.isel({orth_component: orth_index})
+
+            # Need to make sure to use the nearest coordinate that is at least greater than the port bounds
+            hcap_minus = h_cap.sel({h_component: slice(-np.inf, port.bounds[0][port.current_axis])})
+            hcap_plus = h_cap.sel({h_component: slice(port.bounds[1][port.current_axis], np.inf)})
+            hcap_minus = hcap_minus.isel({h_component: -1})
+            hcap_plus = hcap_plus.isel({h_component: 0})
+            # Length of integration along the h_cap contour
+            dcap = (
+                h_coords[port.injection_axis].values[orth_index]
+                - h_coords[port.injection_axis].values[orth_index - 1]
+            )
+
+            h_min_bound = hcap_minus.coords[h_component].values
+            h_max_bound = hcap_plus.coords[h_component].values
+            h_coords_interp = {
+                h_component: np.linspace(
+                    h_min_bound,
+                    h_max_bound,
+                    len(h_coords[port.current_axis] + 2),
+                )
+            }
+            # Integration that corresponds to the tangent H field
+            h_field = h_field.interp(**h_coords_interp)
+            current = h_field.integrate(coord=h_component).squeeze(drop=True)
+
+            # Integration that corresponds with the contribution to current from cap contours
+            hcap_current = (
+                ((hcap_plus - hcap_minus) * dcap).squeeze(drop=True).reset_coords(drop=True)
+            )
+            current = current + hcap_current
+
+            if port.current_axis != (port.voltage_axis + 1) % 3:
+                current *= -1
+            # Return data array of current with coordinates of frequency
+            return current
```

Comment:
If we are to compute impedance from the mode solver result, is the typical computation following the same approach here: compute voltage, current, and then division?

Reply:
You only need the voltage and current to compute the scattering parameters. Computing the impedance would be a separate operation that could be useful though. Using a mode solver result the voltage and current would be computed differently using 2D integrals on the transverse plane.

Reply:
You might need to compute impedance to choose a proper load and/or reference impedance.